packages/components/src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import { SystemProp, Theme } from "@xstyled/styled-components";
+import { Box } from "../../primitives/Box";
+import { useGetImageOrientation } from "./useGetImageOrientation";
+import { Orientation } from "./getImageOrientation";
+import { getInitials } from "./getInitials";
+
+type AvatarSizeOptions = 6 | 8 | 10 | 12 | 14 | 16;
+
+type AvatarProps = {
+  /** Sets the image URL. */
+  src?: string;
+  /** Sets the alternative text. It is used to define initials if src isn't provided or fails to load and to add an accessible label. */
+  alt?: string;
+  /** Sets the avatar size. */
+  size?: AvatarSizeOptions;
+};
+
+const getAvatarSizes = (
+  size: AvatarSizeOptions
+): {
+  w: string;
+  h: string;
+  margin: SystemProp<keyof Theme["space"], Theme>;
+} => {
+  switch (size) {
+    case 6: {
+      return {
+        w: "24px",
+        h: "24px",
+        margin: "space30", // Should be 10px - no token provided
```

Comment:
@richbachman, the design team used a space around the avatar for which we don't have a token. But the margin property only accepts tokens. Which do you think should be our approach here?

Reply:
Let's check in with design and see if the 8px or 12px token will work. I had to do that with the Button component. I don't think they've had chance to go back and edit some of the FIgma files since they removed the 10px spacing.